provider/types.go

```diff
@@ -27,39 +27,52 @@ var extraTypes = map[string]schema.ComplexTypeSpec{
 			Type:        "string",
 			Description: "A Region represents any valid Amazon region that may be targeted with deployments.",
 		},
+		// synced with https://github.com/hashicorp/aws-sdk-go-base/blob/fdf82b33b2d1b0c9165e3b465e0fa17191c6e5b8/internal/endpoints/endpoints.go
 		Enum: []schema.EnumValueSpec{
-			{Value: "af-south-1", Name: "AFSouth1"},
-			{Value: "ap-east-1", Name: "APEast1"},
-			{Value: "ap-northeast-1", Name: "APNortheast1"},
-			{Value: "ap-northeast-2", Name: "APNortheast2"},
-			{Value: "ap-northeast-3", Name: "APNortheast3"},
-			{Value: "ap-south-1", Name: "APSouth1"},
-			{Value: "ap-south-2", Name: "APSouth2"},
-			{Value: "ap-southeast-1", Name: "APSoutheast1"},
-			{Value: "ap-southeast-2", Name: "APSoutheast2"},
-			{Value: "ap-southeast-3", Name: "APSoutheast3"},
-			{Value: "ap-southeast-4", Name: "APSoutheast4"},
-			{Value: "ca-central-1", Name: "CACentral"},
-			{Value: "ca-west-1", Name: "CAWest1"},
-			{Value: "cn-north-1", Name: "CNNorth1"},
-			{Value: "cn-northwest-1", Name: "CNNorthwest1"},
-			{Value: "eu-central-1", Name: "EUCentral1"},
-			{Value: "eu-central-2", Name: "EUCentral2"},
-			{Value: "eu-north-1", Name: "EUNorth1"},
-			{Value: "eu-south-1", Name: "EUSouth1"},
-			{Value: "eu-south-2", Name: "EUSouth2"},
-			{Value: "eu-west-1", Name: "EUWest1"},
-			{Value: "eu-west-2", Name: "EUWest2"},
-			{Value: "eu-west-3", Name: "EUWest3"},
-			{Value: "me-central-1", Name: "MECentral1"},
-			{Value: "me-south-1", Name: "MESouth1"},
-			{Value: "sa-east-1", Name: "SAEast1"},
-			{Value: "us-gov-east-1", Name: "USGovEast1"},
-			{Value: "us-gov-west-1", Name: "USGovWest1"},
-			{Value: "us-east-1", Name: "USEast1"},
-			{Value: "us-east-2", Name: "USEast2"},
-			{Value: "us-west-1", Name: "USWest1"},
-			{Value: "us-west-2", Name: "USWest2"},
+			// aws partition
```

Comment:
In the future we could generate this based on the endpoints listed in the upstream provider.